pkg/controllers/dnspolicy/dns_helper_test.go

```diff
@@ -1023,805 +966,16 @@ func Test_dnsHelper_setEndpoints(t *testing.T) {
-			},
-			probeOne: &v1alpha1.DNSHealthCheckProbe{
-				ObjectMeta: v1.ObjectMeta{
-					Labels: commonDNSRecordLabels(
-						types.NamespacedName{
-							Namespace: "testns",
-							Name:      "testgw"},
-						types.NamespacedName{
-							Namespace: "testns",
-							Name:      "testpolicy",
-						}),
-					Name:      dnsHealthCheckProbeName("1.1.1.1", "testgw", "test"),
-					Namespace: "testns",
-				},
-				Status: v1alpha1.DNSHealthCheckProbeStatus{
-					Healthy: aws.Bool(true),
-				},
-			},
-
-			probeTwo: &v1alpha1.DNSHealthCheckProbe{
-				ObjectMeta: v1.ObjectMeta{
-					Labels: commonDNSRecordLabels(
-						types.NamespacedName{
-							Namespace: "testns",
-							Name:      "testgw"},
-						types.NamespacedName{
-							Namespace: "testns",
-							Name:      "testpolicy",
-						}),
-					Name:      dnsHealthCheckProbeName("1.1.1.1", "testgw", "other"),
-					Namespace: "testns",
-				},
-				Spec: v1alpha1.DNSHealthCheckProbeSpec{
-					FailureThreshold: aws.Int(4),
-				},
-				Status: v1alpha1.DNSHealthCheckProbeStatus{
-					Healthy:             aws.Bool(false),
-					ConsecutiveFailures: 6,
-				},
-			},
-			wantSpec: &v1alpha1.DNSRecordSpec{
-				Endpoints: []*v1alpha1.Endpoint{
-					{
-						DNSName:    "20qri0.lb-0ecjaw.test.example.com",
-						Targets:    []string{"1.1.1.1"},
-						RecordType: "A",
-						RecordTTL:  dns.DefaultTTL,
-					},
-					{
-						DNSName:    "20qri0.lb-0ecjaw.test.example.com",
-						Targets:    []string{"2.2.2.2"},
-						RecordType: "A",
-						RecordTTL:  dns.DefaultTTL,
-					},
-					{
-						DNSName:       "default.lb-0ecjaw.test.example.com",
-						Targets:       []string{"20qri0.lb-0ecjaw.test.example.com"},
-						RecordType:    "CNAME",
-						SetIdentifier: "20qri0.lb-0ecjaw.test.example.com",
-						RecordTTL:     dns.DefaultTTL,
-						ProviderSpecific: []v1alpha1.ProviderSpecificProperty{
-							{
-								Name:  "weight",
-								Value: "120",
-							},
-						},
-					},
-					{
-						DNSName:       "default.lb-0ecjaw.test.example.com",
-						Targets:       []string{"20qri0.lb-0ecjaw.test.example.com"},
-						RecordType:    "CNAME",
-						SetIdentifier: "20qri0.lb-0ecjaw.test.example.com",
-						RecordTTL:     dns.DefaultTTL,
-						ProviderSpecific: []v1alpha1.ProviderSpecificProperty{
-							{
-								Name:  "weight",
-								Value: "120",
-							},
-						},
-					},
-					{
-						DNSName:       "lb-0ecjaw.test.example.com",
-						Targets:       []string{"default.lb-0ecjaw.test.example.com"},
-						RecordType:    "CNAME",
-						SetIdentifier: "default",
-						RecordTTL:     dns.DefaultCnameTTL,
-						ProviderSpecific: []v1alpha1.ProviderSpecificProperty{
-							{
-								Name:  "geo-code",
-								Value: "*",
-							},
-						},
-					},
-					{
-						DNSName:    "test.example.com",
-						Targets:    []string{"lb-0ecjaw.test.example.com"},
-						RecordType: "CNAME",
-						RecordTTL:  dns.DefaultCnameTTL,
-					},
-				},
-			},
-		},
```

Comment:
Lot of test cases being removed here. Are the not still valid? Do these need to be moved/rewritten for `RemoveUnhealthyGatewayAddresses`?

Reply:
yes, they're not valid for setEndpoints anymore but could/should be rewritten for the new function which is doing the logic now. 👍

Reply:
I should say the integration tests required no change and they cover a lot of cases here including multiple listeners. 😄

Reply:
If you think they are covered by integration tests we might not need them, or at least all of them. Some tests for `RemoveUnhealthyGatewayAddresses` might be useful though, and also might be a bit easier to test the results rather than checking all endpoints in a DNSRecord.

Reply:
i pushed a set of tests for that function. Because there's no return it's really just a coverage test and would only really identify if the function changes that there were some issues with nils for example. It's making me think that the call for this could be done as part of the Construction of the GatewayTarget. wdyt?

Reply:
The test can still test the  `MultiClusterGatewayTarget` is in the expected state after calling the function even if it returns nothing. This would be the same if you did it via a constructor or not.

TestCase:
1. Create the MultiClusterGatewayTarget 
 ```
mgt := &MultiClusterGatewayTarget{
				Gateway:               tt.fields.Gateway,
				ClusterGatewayTargets: tt.fields.ClusterGatewayTargets,
				LoadBalancing:         tt.fields.LoadBalancing,
			}
```
3. Call RemoveUnhealthyGatewayAddresses
```
mgt.RemoveUnhealthyGatewayAddresses(tt.args.probes, tt.args.listener)
```
5. Check  MultiClusterGatewayTarget.ClusterGatewayTargets are what you expect them to be
```
if mgt.ClusterGatewayTargets != testCase.want { <- This will need to actually check the equality of the arrays
   t.Errorf("got = %v, want %v", got, testCase.want)
}
```

Reply:
facepalm. thanks. 👍